genie-web/src/main/java/com/netflix/genie/web/tasks/leader/ClusterCheckerTask.java

```
@@ -109,70 +121,99 @@ public ClusterCheckerTask(
-                    final Set<Job> jobs = this.jobSearchService.getAllActiveJobsOnHost(host);
-                    jobs.forEach(
-                        job -> {
-                            try {
-                                this.jobPersistenceService.setJobCompletionInformation(
-                                    job.getId(),
-                                    JobExecution.LOST_EXIT_CODE,
-                                    JobStatus.FAILED,
-                                    "Genie leader can't reach node running job. Assuming node and job are lost."
-                                );
-                                this.lostJobsCounter.increment();
-                            } catch (final GenieException ge) {
-                                log.error("Unable to update job {} to failed due to exception", job.getId(), ge);
-                                this.unableToUpdateJobCounter.increment();
-                            }
-                        }
-                    );
+    private boolean isNodeHealthy(final String host) {
+        //
+        // A node is valid and healthy if all health endpoints excluding the ones mentioned in ignoreHealthEndPoints
+        // are UP.
+        //
+        boolean result = true;
+        try {
+            restTemplate.getForObject(this.scheme + host + this.healthEndpoint, String.class);
+        } catch (final HttpStatusCodeException e) {
+            log.error("Failed validating host {}", host, e);
+            try {
+                final Map<String, Object> responseMap = mapper.readValue(e.getResponseBodyAsByteArray(),
+                    TypeFactory.defaultInstance().constructMapType(Map.class, String.class, Object.class));
+                for (Map.Entry<String, Object> responseEntry : responseMap.entrySet()) {
+                    if (responseEntry.getValue() instanceof Map
+                        && !ignoreHealthEndPoints.contains(responseEntry.getKey())
```

I'd put this condition first so it fails fast before even bothering checking the instance type